rb/lib/selenium/webdriver/remote/bridge.rb

```diff
@@ -605,11 +552,58 @@ def find_elements_by(how, what, parent = nil)
           ids.map { |id| Element.new self, element_id_from(id) }
         end
 
+        protected
+
+        def capabilities_class
+          W3CCapabilities
+        end
+
+        def default_capabilities
+          capabilities_class.firefox
+        end
+
         private
 
-        def assert_javascript_enabled
-          return if capabilities.javascript_enabled?
-          raise Error::UnsupportedOperationError, 'underlying webdriver instance does not support javascript'
+        def bridge_module
+          Module.nesting[1]
+        end
+
+        def process_capabilities(opts)
+          capabilities = opts.delete(:desired_capabilities) || default_capabilities
+          capabilities = Capabilities.send(capabilities) if capabilities.is_a?(Symbol)
```

Comment:
Ah, the `Firefox::Bridge#process_capabilities` method is intended to handle the things that are local only, and the `Capabilities#self.firefox` method is intended to handle the things that are applicable regardless of local or remote.

Before this PR is merged, I'm planning on revamping Watir browser configurations. I'm going to be ok with telling Watir users that if they want to use selenium-webdriver 3.1.x then they need to use watir 6.4 or whatever, and change their parameters as necessary.